src/hal/src/pso/mod.rs

```diff
@@ -21,6 +21,9 @@ pub enum CreationError {
     /// Unknown other error.
     #[error("Implementation specific error occurred")]
     Other,
+    /// Shader module creation error.
+    #[error("Shader creation failed: {0:}")]
+    ShaderCreationError(String),
```

Comment:
I think we always know the shader stage when we issue this error, don't we? In this case, we should make it `ShaderCreationError(Stage, String)`. Would you be open to try it out?